hexrd/ui/calibration_crystal_slider_widget.py

```diff
@@ -0,0 +1,173 @@
+from enum import IntEnum
+
+from PySide2.QtCore import QObject, QSignalBlocker, Signal
+from PySide2.QtWidgets import QProxyStyle, QStyle
+
+from hexrd.ui.hexrd_config import HexrdConfig
+from hexrd.ui.ui_loader import UiLoader
+
+
+class SpinBoxStyle(QProxyStyle):
+    def styleHint(self, hint, option=None, widget=None, returnData=None):
+        """Increase the auto-repeat threshold to 1 sec.
+
+        Otherwise single click causes two increments.
+        """
+        if hint == QStyle.SH_SpinBox_ClickAutoRepeatThreshold:
+            return 1000
+        else:
+            return super().styleHint(hint, option, widget, returnData)
+
+
+class WidgetMode(IntEnum):
+    ORIENTATION = 1  # sliders update orientation
+    POSITION = 2     # sliders update position
+
+
+class CalibrationCrystalSliderWidget(QObject):
+    changed = Signal(int, int, float)
+
+    # Conversions from configuration value to slider value and back
+    CONF_VAL_TO_SLIDER_VAL = 10
+    SLIDER_VAL_TO_CONF_VAL = 0.1
+
+    def __init__(self, parent=None):
+        super(CalibrationCrystalSliderWidget, self).__init__(parent)
+        self._mode = WidgetMode.ORIENTATION
+
+        loader = UiLoader()
+        self.ui = loader.load_file(
+            'calibration_crystal_slider_widget.ui', parent)
+        for w in self.spinbox_widgets:
+            w.setStyle(SpinBoxStyle())
+
+        self._orientation = [0.0] * 3
+        self._orientatin_range = 30.0
```

Comment:
Let pull the default range into a constant

Reply:
Good idea. Just pushed commit with that change.